numpy/core/src/multiarray/mapping.c

```diff
@@ -2452,6 +2345,10 @@ mapiter_fill_info(PyArrayMapIterObject *mit, npy_index_info *indices,
             mit->fancy_dims[j] = 1;
             /* Does not exist */
             mit->iteraxes[j++] = -1;
+            if (mit->dimensions[mit->nd_fancy - 1] > 1) {
```

Comment:
That test I guessed the result wrong... If its a True, its like a dimension of 1 and that means it can broadcast:
```
if ((indices[i].value == 0) && (mit->dimensions[mit->nd_fancy - 1] > 1)) {
```
should fix it. Wonky stuff mixing boolean arrays with other indexes....